### PR TITLE
Update from Base and recommit changes in Pkg.jl on top of it.

### PR DESCRIPTION
**Pull request by KristofferC**
The Pkg.jl repo and Base Pkg has slightly diverged. This merges the history by adding a commit that simply sets the file to the version of Base at 1.0 and then recommits the changes in the repo on top of it.

### CODE REVIEW

File: src/Types.jl [hunk 1]
```diff
@@ -908,8 +914,19 @@ end
 function registries(; clone_default=true)::Vector{String}
     isempty(depots()) && return String[]
     user_regs = abspath(depots1(), "registries")
+    # TODO: delete the following let block in Julia 1.0
```
Comment (fredrikekre) [inline src/Types.jl]: rm?

File: src/resolve/FieldValues.jl [hunk 2]
```diff
@@ -72,9 +72,9 @@ Base.copy(a::FieldValue) = FieldValue(a.l0, copy(a.l1), copy(a.l2), a.l3)
 # some hard constraint is being violated
 validmax(a::FieldValue) = a.l0 >= 0
 
-# like usual indmax, but favors the highest indices
+# like usual argmax, but favors the highest indices
 # in case of a tie
-function Base.indmax(f::Field)
+function Base.argmax(f::Field)
```
Comment (fredrikekre) [inline src/resolve/FieldValues.jl]: Maybe no reason to extend this since it is only used internally

File: test/pkg.jl [hunk 3]
```diff
@@ -345,6 +346,8 @@ temp_pkg_dir() do project_path
     end
 end
 
+# This test seems to error on AppVeyor so disable for now
```
Comment (fredrikekre) [inline test/pkg.jl]: Still true?

Comment (KristofferC) [reply test/pkg.jl]: Yes

File: src/Types.jl (hunk 1 above)
Comment (KristofferC) [reply src/Types.jl]: Can removing it in a while.

**Pull request merged**